client/src/components/LanguageGroups.js

```diff
@@ -13,9 +13,12 @@ export default () => {
         setValue(nextValue)
       }}
     >
-      {groups.map((language) => (
-        <Radio value={language.code}>{language.language}</Radio>
-      ))}
+      {groups.map(
+        (language) => (
+          language.code == props.code && props.setIsTextDetected(true),
+          (<Radio value={language.code}>{language.language}</Radio>)
+        )
+      )}
```

Comment:
```suggestion
      {groups.map(
        (group) => <Radio value={group.code}>{group.language}</Radio>
      )}
```
Text detection judgment shall be done outside Radio Group component.
Use `===` instead of `==`